### ISSUE TITLE
link for Media Player Framework for iOS is broken

### ISSUE THREAD

**Issue by anavgale** [Azure/azure-content#4037]
here as well as many Azure docs the link for Media Player Framework for iOS is broken. What is the correct link? Couldn't find.

**Comment by deneha** [comment]
@anavgale Thanks for raising this issue.  What is the specific page where you experience this issue?

**Comment by anavgale** [comment]
1. Go to this link - https://azure.microsoft.com/en-in/documentation/articles/media-services-develop-video-players/

Under the section - "Other Tools for Creating Player Applications" click on - Media Player Framework for iOS - which is broken.

2. Same URL is also referenced at - https://channel9.msdn.com/Series/Windows-Azure-Media-Services-Tutorials/An-introduction-to-Azure-Media-Player-Framework-for-IOS
Which is broken.

Basically, I am searching for - How can I build Video player for iOS to consume content from Azure Media Services? These  video talks about the same but link given in the description of the video is missing.

Can you please point me to the correct content?

Thanks.

**Comment by deneha** [comment]
@anavgale Thank you!  I'll contact the author for you.

**Comment by Juliako** [comment]
The link in the https://azure.microsoft.com/en-in/documentation/articles/media-services-develop-video-players/ topic was removed. The topic now contains comments that address the iOS framework question. I will incorporate the comments into the topic later.

**Comment by deneha** [comment]
@anavgale Does this resolve your issue?  Please let us know if you need further assistance by replying to this issue.  We'll close this in a couple days if we don't hear back.

**Comment by ajaynavgale** [comment]
Sorry I am late to reply. 

Actually I am still looking for an answer. 

Can I play AES encrypted and protected media form Azure Media Services using Azure Media Player on iOS app using webview?

Thanks.

**Comment by anavgale** [comment]
Sorry I am late to reply.

Actually I am still looking for an answer.

Can I play AES encrypted and protected media form Azure Media Services using Azure Media Player on iOS app using webview?

Thanks.

**Comment by rajputam** [comment]
When using the HTML5 video tag in Safari, has no APIs to set a token. If you just attach the token to the player, for Safari HLS, Players will pass down Safari to handle the playback and Safari will ignore the token. This is specifically Apple’s implementation and we have no control over it.  

When using webview, which will use the HTML5 video tag with Apple today, in order to playback AES encrypted assets with a token, you will need to create a proxy in order to append the token into the HLS manifest. If, for some reason you don't need a token, then this proxy is not required. Check out this blog for more details: http://azure.microsoft.com/blog/2015/03/06/how-to-make-token-authorized-aes-encrypted-hls-stream-working-in-safari/ 

This proxy implementation is a limitation of all players using Apple's HTML5 native video tag, including just using the video tag in browser from Apple.  This limitation should go away once Apple starts to support MSE in their browsers for iOS -> though older devices/browser versions will still be affected.

**Comment by deneha** [comment]
@Juliako Could you please review this question?

**Comment by Juliako** [comment]
Hi Anavgale, please let us know if your question has been answered.
thank you,
Julia

**Comment by ajaynavgale** [comment]
yes, thank you. I am referring to the blog mentioned and implementing the proxy. Thanks a lot for your help.

**Comment by Juliako** [comment]
Great! I am glad you were able to make it work.

**Comment by deneha** [comment]
We're closing this issue now since the problem seems resolved.  Please comment if we should reopen.
